safe/trainer/collator.py

```diff
@@ -86,7 +85,7 @@ def __call__(self, samples: List[Union[List[int], Any, Dict[str, Any]]]):
             )
         else:
             batch = tokenizer.pad(
-                examples,
+                samples,
```

Comment:
pad will fail if you are not using a copy of samples on which you pop the unused key. 

It's designed to either tokenize on the fly or pad a pretokenized data. If the data is pretokenized your changes will raise an error.